Lib/WindowsRuntime/Blob/CloudBlobClient.cs

```diff
@@ -559,7 +559,7 @@ private RESTCommand<ServiceStats> GetServiceStatsImpl(BlobRequestOptions request
             retCmd.BuildRequest = (cmd, uri, builder, cnt, serverTimeout, ctx) => BlobHttpRequestMessageFactory.GetServiceStats(uri, serverTimeout, ctx, this.GetCanonicalizer(), this.Credentials);
             retCmd.RetrieveResponseStream = true;
             retCmd.PreProcessResponse = (cmd, resp, ex, ctx) => HttpResponseParsers.ProcessExpectedStatusCodeNoException(HttpStatusCode.OK, resp, null /* retVal */, cmd, ex);
-            retCmd.PostProcessResponse = (cmd, resp, ctx) => Task.Factory.StartNew(() => BlobHttpResponseParsers.ReadServiceStats(cmd.ResponseStream));
+            retCmd.PostProcessResponse = (cmd, resp, ctx) => Task.FromResult(BlobHttpResponseParsers.ReadServiceStats(cmd.ResponseStream));
```

Comment:
This comment applies to all cases of PostProcessResponse methods touched by this change where the internal logic for instance ReadServiceStats does a synchronous read on a network stream where this change will lead to blocking the calling thread until the read is completed. Spinning up an extra thread here, although not a good solution, could still be considered better than blocking.

The good news is, we have a set of changes which is going to fix this issue by making the PostProcess actually async to enable us take async all the way down in the parsers.

So we suggest to hold off on this section of the changes and we will address this with our pending set of changes to make everything async all the way through. I will create an issue to track this task.

Reply:
@erezvani1529 Please allow me to kindly disagree here for a moment. If you look into the executor class workflow you will see that  `PostProcessResponse` func is called after several asynchronous continuations have already been scheduled and run. For example `await client.SendAsync` comes first on the HttpClient and then `await executionState.Resp.Content.ReadAsStreamAsync()` on the `HttpContent` and I only took into account the happy path. Since we consistently apply `ConfigureAwait(false)` the continuations will be rescheduled on the `TaskScheduler.Default` and can freely use the ThreadPool available. So there is no blocking happing and it is perfectly valid to execute synchronous blocking code in a continuation. In the future, if the truly async reading there is supported the code would be even higher optimized. But rescheduling unnecessary again to the thread pool explicitly inside a deep continuation of an already asynchronous call stack is only putting unnecessary pressure on the threadpool without achieving much.

Reply:
An additional comment. I get that for example the streaming API in .NET Core uses the value task. So in case the stream is already materialized it could synchronously complete. But even in that case you'd had other continuations scheduled before. 

https://twitter.com/davidfowl/status/750580955564367872

Reply:
Good point, this was my concern, but you're right, we should no longer be on the calling thread.  Thanks for doing all this!